### PR TITLE
feat(Menu): Add children api support

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Adding Children API support to Menu

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -215,6 +229,12 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
 
     const slotProps = composeOptions.resolveSlotProps<MenuItemProps & MenuItemState>({
       ...props,
+      active,
```
Comment (jurokapsiar) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:232]: shouldn't we also resolve `accessibility` so that child behavior is applied correctly? If I use `tabBeavior` I would expect the items to be `tab`s, just like in shorthand

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:239]: Done, please check if it's the proper way :)

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -427,7 +465,7 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
               targetRef={itemRef}
               {...getPopperPropsFromShorthand(menu)}
             >
-              {Menu.create(menu, {
+              {createShorthand(parentProps.__parentComponent || Menu, menu, {
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:468]: If we want to support custom components for the submenus, we may use here `composeOptions.slots.menu` as a first option... But to be honest, for now would keep it simple

File: packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx [hunk 3]
```diff
@@ -29,24 +32,53 @@ const MenuitemIndicatorSaturated = compose(MenuItemIndicator, {
...
+
+    content: MenuItemContentPurple,
+
     indicator: MenuitemIndicatorSaturated,
+
     wrapper: MenuItemWrapperDashed,
   },
 });
 
+const MenuCoral = compose<'ul', MenuProps, MenuStylesProps, MenuProps, MenuStylesProps>(Menu, {
```
Comment (jurokapsiar) [inline packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx:47]: @mnajdova, @layershifter does the consumer need to know that Menu renders `ul`? Should not we expose it and have something like `Menu.defaultElementType`?

File: packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDividerHorizontal.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { Menu } from '@fluentui/react-northstar';
+import { TriangleEndIcon } from '@fluentui/react-icons-northstar';
+
+const MenuExampleKind = () => (
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDividerHorizontal.tsx:5]:
```suggestion
const MenuExampleDividerHorizontal = () => (
```

File: packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleSlot.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import { Menu } from '@fluentui/react-northstar';
+
+const MenuExample = () => (
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleSlot.tsx:4]:
```suggestion
const MenuExampleSlot = () => (
```

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { Menu, tabListBehavior } from '@fluentui/react-northstar';
+
+class MenuExampleTabShorthand extends React.Component {
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx:4]: Let's convert it to FC

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  VideoCameraEmphasisIcon,
+  SettingsIcon,
+  MoreIcon,
+  BookmarkIcon,
+  TranslationIcon,
+  TrashCanIcon,
+  MarkAsUnreadIcon,
+} from '@fluentui/react-icons-northstar';
+
+const MenuExampleToolbarShorthand = () => (
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx:18]:
```suggestion
const MenuExampleToolbar = () => (
```

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuControlled.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { Menu, Header } from '@fluentui/react-northstar';
+
+class MenuWithSubmenuControlledExample extends React.Component {
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuControlled.tsx:4]: Let's convert it to FC and use knobs

File: packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+const MenuExampleVertical = () => (
+  <Menu primary defaultActiveIndex={0} vertical>
+    <Menu.Item
+      index={0}
+      icon={
+        <BookmarkIcon
+          {...{
+            outline: true,
+          }}
+        />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx:14]:
```suggestion
        <BookmarkIcon outline />
```

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  MoreIcon,
+  BookmarkIcon,
+  TranslationIcon,
+  TrashCanIcon,
+  MarkAsUnreadIcon,
+} from '@fluentui/react-icons-northstar';
+
+const MenuExampleToolbar = () => (
+  <Menu defaultActiveIndex={0} iconOnly accessibility={menuAsToolbarBehavior} aria-label="Compose Editor">
+    <Menu.Item index={0} aria-label="Format Tool" icon={<FormatIcon outline={true} />} />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx:20]:
```suggestion
    <Menu.Item index={0} aria-label="Format Tool" icon={<FormatIcon outline />} />
```

And the same for lines below

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenu.tsx [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+import { CircleIcon } from '@fluentui/react-icons-northstar';
+
+const MenuExampleWithSubMenu = () => (
+  <Menu defaultActiveIndex={0}>
+    <Menu.Item
+      index={0}
+      menu={
+        <>
+          <Menu.Item>
+            <Menu.ItemContent key="item1">item1</Menu.ItemContent>
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenu.tsx:12]:
```suggestion
            <Menu.ItemContent>item1</Menu.ItemContent>
```

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenu.tsx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+      }
+      content="Editorials"
+    />
+
+    <Menu.Item
+      index={1}
+      menu={
+        <>
+          <Menu.Item>
+            <Menu.ItemContent key="item1">item1</Menu.ItemContent>
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenu.tsx:44]:
```suggestion
            <Menu.ItemContent>item1</Menu.ItemContent>
```

File: packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+      }
+      content="Editorials"
+    />
+    <Menu.Item index={1} content="Reviews" icon={<WordIcon {...{}} />} />
+    <Menu.Item index={2} content="Upcoming Events" />
+    <Menu.Item
+      index={3}
+      icon={<CalendarIcon {...{}} />}
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx:18]:
```suggestion
      icon={<CalendarIcon />}
```

File: packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+      icon={
+        <BookmarkIcon
+          {...{
+            outline: true,
+          }}
+        />
+      }
+      content="Editorials"
+    />
+    <Menu.Item index={1} content="Reviews" icon={<WordIcon {...{}} />} />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx:14]:
```suggestion
    <Menu.Item index={1} content="Reviews" icon={<WordIcon />} />
```

File: packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+    <Menu.Item index={2} content="Upcoming Events" />
+    <Menu.Item
+      index={3}
+      icon={<CalendarIcon {...{}} />}
+      content="View full calendar with content so long that it wraps"
+      menu={
+        <>
+          <Menu.Item>
+            <Menu.ItemContent key="item1">item1</Menu.ItemContent>
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx:23]:
```suggestion
            <Menu.ItemContent>item1</Menu.ItemContent>
```

File: packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+          </Menu.Item>
+          <Menu.Item>
+            <Menu.ItemContent key="item2">item2</Menu.ItemContent>
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Variations/MenuExampleVerticalPrimary.tsx:26]:
```suggestion
            <Menu.ItemContent>item2</Menu.ItemContent>
```

File: packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx [hunk 17]
```diff
@@ -29,24 +32,53 @@ const MenuitemIndicatorSaturated = compose(MenuItemIndicator, {
...
   displayName: 'MenuItemBlue',
+
   slots: {
     icon: MenuItemIconGreen,
-    content: MenuItemContentPink,
+
+    content: MenuItemContentPurple,
+
     indicator: MenuitemIndicatorSaturated,
+
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx:42]: nit: Let's remove added newlines

File: packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx (hunk 3 above)
Comment (layershifter) [reply packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx:47]: It's questionable, I don't think so

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 18]
```diff
@@ -196,13 +198,20 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
-      onClick: (e, itemProps) => {
+    const handleClick = React.useCallback(
+      (e, itemProps) => {
         const { index } = itemProps;
 
         setActiveIndex(e, index);
 
         _.invoke(props, 'onItemClick', e, itemProps);
+      },
+      [props, setActiveIndex],
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:209]: This will not work as `props` will change during each render

```tsx
function useActualProps<P>(props: P) {
  const actualProps = React.useRef<P>(props)

  React.useEffect(() => {
    actualProps.current = props
  })

  return actualProps
}

const actualProps = useActualProps(props)
```

```suggestion
        _.invoke(actualProps.current, 'onItemClick', e, itemProps);
      },
      [actualProps, setActiveIndex],
```

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 19]
```diff
@@ -256,6 +265,22 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
+      onItemClick: handleClick,
+      variables,
+      pointing,
+      primary,
+      underlined,
+      iconOnly,
+      vertical,
+      secondary,
+      pills,
+      inSubmenu: props.submenu,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:279]:
```suggestion
      inSubmenu: props.submenu,
```

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 20]
```diff
@@ -256,6 +265,22 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
+      accessibility: accessibility && accessibility(props).childBehaviors,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:280]: I don't think that this object is stable

File: packages/fluentui/react-northstar/src/components/Menu/menuContext.ts [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+  variables: ComponentVariablesInput;
+  pointing: boolean | 'start' | 'end';
+  primary: boolean;
+  underlined: boolean;
+  iconOnly: boolean;
+  vertical: boolean;
+  secondary: boolean;
+  pills: boolean;
+  inSubmenu: boolean;
+  onItemClick?: (e: React.KeyboardEvent | React.MouseEvent, itemIndex: MenuItemProps) => void;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/menuContext.ts:18]:
```suggestion
  onItemClick?: (e: React.KeyboardEvent | React.MouseEvent, itemIndex: number) => void;
```

File: packages/fluentui/react-northstar/src/components/Menu/menuContext.ts [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+    | 'primary'
+    | 'pointing'
+    | 'underlined'
+    | 'iconOnly'
+    | 'vertical'
+    | 'inSubmenu'
+    | 'pills'
+    | 'secondary'
+    | 'accessibility'
+    | '__parentComponent'
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/menuContext.ts:38]: I would expect that we will have a separate context for this as I see we need under it at least: `menu`, `item`, `divider`

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 23]
```diff
@@ -256,6 +265,22 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
       });
     };
 
+    const childProps: MenuContextValue = {
+      activeIndex: +activeIndex,
+      onItemClick: handleClick,
+      variables,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:271]: This is a bit scary: as these props are related to design and can exist only under `compose`...

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx (hunk 20 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:280]: Hm?

File: packages/fluentui/react-northstar/src/components/Menu/menuContext.ts (hunk 22 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/menuContext.ts:38]: Not sure if I get what you mean.. @mnajdova Maybe have a idea here if you are referring to the `__parentComponent`.

Comment (mnajdova) [reply packages/fluentui/react-northstar/src/components/Menu/menuContext.ts:38]: Yeah, we can do different context, as it's needed only in Menu & MenuItem, not divider

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/menuContext.ts:38]: Ok

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/menuContext.ts:38]: Should I create 2 different context ( one for item and one for divider ) and use like 
 

```jsx
<MenuDividerContextProvider value={itemProps}>
     <MenuItemContextProvider value={itemProps}>
         {childrenExist(children) ? children : renderItems()}
     </MenuItemContextProvider>
</MenuDividerContextProvider>
```

 
Or you meant to split the Menu context and use something like:
 

```jsx
 <MenuContextProvider value={{ item: itemProps, divider: dividerProps }}>
     {childrenExist(children) ? children : renderItems()}
 </MenuContextProvider>
```

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuControlled.tsx (hunk 8 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuControlled.tsx:4]: Done

File: packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx (hunk 17 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx:42]: Done

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx (hunk 6 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx:4]: Done

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx (hunk 18 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:212]: There's a specific better place to me to create this `useActualProps` or should I create it here in this file itself?

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.shorthand.tsx [hunk 24]
```diff
@@ -7,19 +7,17 @@ const items = [
...
-        underlined
-        primary
-        accessibility={tabListBehavior}
-        aria-label="Today's events"
-      />
-    );
-  }
-}
+const MenuExampleTabShorthand: React.FC = props => {
+  return (
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.shorthand.tsx:11]: _nit_: we don't need `return` statement here

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx (hunk 18 above)
Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:212]: For now: let's add it here

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx (hunk 20 above)
Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:283]: Every time when `accessibility(props)` will be evaluated it will return new `childBehaviors` => all children will be notified => rerender everything 💣

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+      icon={<MoreIcon outline={true} />}
+      menu={
+        <>
+          <Menu.Item index={7} aria-label="item 1" content="item 1" icon={<BookmarkIcon outline />} />
+          <Menu.Divider />
+          <Menu.Item index={9} aria-label="item 2" content="item 2" icon={<MarkAsUnreadIcon outline />} />
+          <Menu.Item index={10} aria-label="item 3" content="item 3" icon={<TranslationIcon outline />} />
+          <Menu.Divider />
+          <Menu.Item index={10} aria-label="item 4" content="item 4" icon={<TrashCanIcon outline />} />
+        </>
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx:40]: This part looks like complete misuse 🤔 It should be something like this:

```jsx
<Menu defaultActiveIndex={0} iconOnly accessibility={menuAsToolbarBehavior} aria-label="Compose Editor">
  <Menu.Item index={0} aria-label="Format Tool" icon={<FormatIcon outline />} />
  <Menu.Item>
     <Menu.Item index={7} aria-label="item 1" content="item 1" icon={<BookmarkIcon outline />} />
     <Menu.Divider />
  </Menu.Item>
</Menu>
```

Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx:40]: For shorthand we have an array of `MenuItem` should it be different and be and `Menu` component here? I guess for that we would have to change the `Menu.create` inside `MenuItem`

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+  TrashCanIcon,
+  MarkAsUnreadIcon,
+} from '@fluentui/react-icons-northstar';
+
+const MenuExampleToolbar = () => (
+  <Menu defaultActiveIndex={0} iconOnly accessibility={menuAsToolbarBehavior} aria-label="Compose Editor">
+    <Menu.Item index={0} aria-label="Format Tool" icon={<FormatIcon outline />} />
+    <Menu.Item index={1} aria-label="Paperclip Tool" icon={<PaperclipIcon outline={true} />} />
+    <Menu.Item index={2} aria-label="Emoji Tool" disabled icon={<EmojiIcon outline={true} />} />
+    <Menu.Item index={3} aria-label="Giphy tool" icon={<GiphyIcon outline={true} />} />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx:23]:
```suggestion
    <Menu.Item index={3} aria-label="Giphy tool">
      <Menu.ItemIcon><GiphyIcon outline /></Menu.ItemIcon>
    </Menu.Item>
```

File: packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDivider.tsx [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+const MenuExampleKind = () => (
+  <Menu defaultActiveIndex={0} vertical pointing="start">
+    <Menu.Item index={0}>
+      <Menu.ItemContent>Editorials</Menu.ItemContent>
+    </Menu.Item>
+    <Menu.Divider />
+    <Menu.Item index={1}>
+      <Menu.ItemContent>Reviews</Menu.ItemContent>
+    </Menu.Item>
+    <Menu.Divider content="..." />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDivider.tsx:13]:
```suggestion
    <Menu.Divider>...</Menu.Divider>
```

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 28]
```diff
@@ -259,6 +268,22 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
+      pointing,
+      primary,
+      underlined,
+      iconOnly,
+      vertical,
+      secondary,
+      pills,
+      inSubmenu: props.submenu,
+      accessibility: accessibility && accessibility(props).childBehaviors,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:283]:
```suggestion
      // TODO: please rework me
      accessibilityBehaviorForItem: accessibility && accessibility(props).childBehaviors.item,
      accessibilityBehaviorForDivider: accessibility && accessibility(props).childBehaviors.divider,
```

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 29]
```diff
@@ -178,28 +181,46 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
...
+      underlined = parentProps.underlined,
+      iconOnly = parentProps.iconOnly,
+      inSubmenu = parentProps.inSubmenu,
+      pills = parentProps.pills,
+      pointing = parentProps.pointing,
       className,
       design,
       styles,
       variables,
+      accessibility = parentProps.accessibility.item,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:223]:
```suggestion
      accessibility = parentProps.accessibility.item || menuItemBehavior,
```

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 30]
```diff
@@ -585,7 +622,7 @@ MenuItem.propTypes = {
 
 MenuItem.defaultProps = {
   as: 'a',
-  accessibility: menuItemBehavior as Accessibility,
+  accessibility: menuItemBehavior,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:625]:
```suggestion
```

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.shorthand.tsx (hunk 24 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.shorthand.tsx]: Done

File: packages/fluentui/react-northstar/src/components/Menu/menuContext.ts (hunk 22 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/menuContext.ts]: Reverted

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx (hunk 25 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleToolbar.tsx]: Reverted examples with submenu for now as agreed

File: packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDivider.tsx (hunk 27 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDivider.tsx]: done

File: packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDividerHorizontal.tsx [hunk 31]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { Menu } from '@fluentui/react-northstar';
+import { TriangleEndIcon } from '@fluentui/react-icons-northstar';
+
+const MenuExampleDividerHorizontal = () => (
+  <Menu defaultActiveIndex={0} underlined>
+    <Menu.Item index={0}>
+      <Menu.ItemContent>Editorials</Menu.ItemContent>
+    </Menu.Item>
+    <Menu.Divider>"▸"</Menu.Divider>
```
Comment (mnajdova) [inline packages/fluentui/docs/src/examples/components/Menu/Slots/MenuExampleDividerHorizontal.tsx:10]:
```suggestion
    <Menu.Divider>▸</Menu.Divider>
```

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx (hunk 6 above)
Comment (mnajdova) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx:4]: FC?

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx (hunk 29 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx]: Done

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx (hunk 6 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleTabList.tsx]: Done

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx (hunk 18 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/Menu.tsx]: Done

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 32]
```diff
@@ -199,13 +213,20 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
+    const handleClick = React.useCallback(
+      (e, itemProps) => {
         const { index } = itemProps;
 
         setActiveIndex(e, index);
 
-        _.invoke(props, 'onItemClick', e, itemProps);
+        _.invoke(actualProps.current, 'onItemClick', e, itemProps);
+      },
+      [actualProps, setActiveIndex],
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:224]: Actually here we don't need all props, we need only `onItemClick` from the props. You can replace the `_.invoke` with 
```
onItemClick && onItemClick(e, itemProps);
```

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 33]
```diff
@@ -150,12 +161,15 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
       design,
       secondary,
+      accessibility,
     } = props;
 
     const ElementType = getElementType(props);
     const slotProps = composeOptions.resolveSlotProps<MenuProps>(props);
     const unhandledProps = useUnhandledProps(composeOptions.handledProps, props);
 
+    const actualProps = useActualProps(props);
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:171]: See comment below, we will not need this after all

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 34]
```diff
@@ -259,6 +280,25 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
+      variables,
+      pointing,
+      primary,
+      underlined,
+      iconOnly,
+      vertical,
+      secondary,
+      pills,
+      inSubmenu: props.submenu,
+      // @TODO: please rework me
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:297]:
```suggestion
      // TODO: please rework me
```
nit

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx (hunk 32 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/Menu.tsx]: Done

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 35]
```diff
@@ -215,6 +237,17 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
 
     const slotProps = composeOptions.resolveSlotProps<MenuItemProps & MenuItemState>({
       ...props,
+      accessibility,
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:240]: It will be hard to follow what props should be defined here in the future, which may lead to unexpected bugs. Can we refactor this to something like:
```
...props,
...parentProps,
accessibility,
variables: mergeComponentVariables(variables, parentProps.variables),
isFromKeyboard,
menuOpen,
```

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx [hunk 36]
```diff
@@ -199,13 +202,18 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
...
-    const handleItemOverrides = predefinedProps => ({
-      onClick: (e, itemProps) => {
+    const handleClick = React.useCallback(
+      (e, itemProps) => {
         const { index } = itemProps;
-
         setActiveIndex(e, index);
+        onItemClick && onItemClick(e, itemProps);
+      },
+      [setActiveIndex],
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Menu/Menu.tsx:211]:
```suggestion
      [onItemClick, setActiveIndex],
```

File: packages/fluentui/react-northstar/src/components/Menu/Menu.tsx (hunk 32 above)
Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Menu/Menu.tsx]: If `onItemClick` will be close it will be unstable: `<Menu onItemClick={() => {}} />

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx (hunk 35 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:242]: Done
